core/src/main/scala/Nelson.scala

```diff
@@ -359,13 +359,18 @@ object Nelson {
       StoreOp.findBlueprint(name, revision).foldMap(cfg.storage)
     }
 
+  /*
+   * NOTE: to prevent totally invalid templates being specified (i.e. things that make the
+   * template parser choke) we defensivly try to proof the inbound blueprint which should
+   * raise an error in the event the blueprint is junk.
+   */
   def createBlueprint(name: String, description: Option[String], sha: Sha256, template: String): NelsonK[Option[Blueprint]] =
-    Kleisli { cfg =>
+    proofBlueprint(template).flatMap(_ => Kleisli { cfg =>
```

Comment:
can't this just go into the for comprehension?

Reply:
The for-comp on the right is a `StoreOp`, not a `NelsonK`. At present, the `StoreOp` is lifted into the `NelsonK`, but you wont be able to compose them directly, unless im missing something or there's a different syntax to convert / lift between the two types.

Reply:
right sorry, I was misreading and I don't remember how to scala, lol